parseSchema/index.js

```diff
@@ -0,0 +1,111 @@
+"use strict";
+
+const Parse = require('parse/node');
+
+function updateSchemaOnlyWhenNecessary(existingSchemaData, intendedSchema)
+{
+  // remove any existing fields from the request.
+  Object.getOwnPropertyNames(existingSchemaData.fields).forEach(existingField => { delete intendedSchema._fields[existingField] ; });
```

Comment:
I'm worried about what happens when an existing field is not present in the fields array in the intendedSchema, does an error get thrown when it tries to access an element that doesn't exist?

Reply:
no... the code here doesn't delete and we will never delete any fields.  What's happening in this function is for any field that is already existing, we remove it **from the request**.    A request doesn't have to include all the fields.  A request that attempts to add an existing field will cause an exception, and that's what this function tries to avoid.